extensions/amp-story/1.0/amp-story-quiz.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-story-quiz-1.0.css';
+import {createShadowRootWithStyle} from '../0.1/utils';
+import {htmlFor} from '../../../src/static-template';
+import {isLayoutSizeDefined} from '../../../src/layout';
+
+/** @private {Array<string>} */
+const answerChoiceOptions = ['A', 'B', 'C', 'D'];
+
+export class AmpStoryQuiz extends AMP.BaseElement {
+  /**
+   * @param {!AmpElement} element
+   */
+  constructor(element) {
+    super(element);
+
+    /** @private {?ShadowRoot} */
+    this.shadowRoot_ = null;
+
+    /** @private {boolean} */
+    this.hasReceivedResponse_ = false;
+  }
+
+  /** @override */
+  buildCallback() {
+    const html = htmlFor(this.element);
+    const shadowElement = html`
+      <div class="i-amp-story-quiz-container">
+        <div class="i-amp-story-quiz-prompt-container"></div>
+        <div class="i-amp-story-quiz-option-container"></div>
+      </div>
+    `;
+
+    this.shadowRoot_ = createShadowRootWithStyle(
+      this.element,
+      shadowElement,
+      CSS
+    );
+
+    // TODO: CONVERT THIS TO STANDARD AMP ERROR REPORTING
+    this.attachContent_(e => {
+      console.log('error!', e);
+      throw new Error(e);
+    });
+    this.attachOptionActionHandlers_();
+
+    // TODO: ATTACH SURFACE OPTIONS USING CSS CUSTOM PROPERTIES
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /**
+   * @private
+   * Finds the prompt and options content
+   * and adds it to the shadow DOM.
+   *
+   * @param {Function} handleError
+   */
+  attachContent_(handleError) {
+    // TODO: If prompt is optional I need to handle the case where it's not present
+    const prompt = this.element.children[0];
+    // First child must be heading h1-h3
+    if (
+      !(prompt instanceof HTMLHeadingElement) ||
+      prompt.tagName[prompt.tagName.length - 1] > 3
```

Comment:
Can we check the `tagName` instead? This might be easier to read, and `instanceof` can lead to some very obscure bugs ([example](https://github.com/ampproject/amphtml/pull/21972#discussion_r278343141)).